eta/core/data.py

```diff
@@ -827,9 +827,31 @@ def cull(self, field, keep_values=None, remove_values=None):
                 "Either keep_values or remove_values must be provided")
 
         # Cull records
-        inds = []
+        inds = set()
         for v in keep_values:
-            inds += lud[v]
+            inds.update(lud[v])
+        self.keep_inds(inds)
+
+        return len(self)
+
+    def cull_with_function(self, field, func=lambda x: x):
```

Comment:
why provide a default value here? doesn't make sense to call this method without providing a function.

We are not one of those shops that requires almost every arg to have default values so that we can "silently" deprecate function syntaxes without breaking existing code. I generally like implementations that minimize the number of keystrokes I make ;)

Reply:
Fixed.